File: example/Calculator.ts
/* eslint-disable @typescript-eslint/no-explicit-any */
export class Calculator {
  private display: string = "";
  private values: { [key: string]: any } = {};
  private id: number = Math.random(); // Add a unique ID to track instances

  press(button: string): Calculator {
    this.display = this.display + button;
    return this;
  }

  enter(): void {
    try {
      // Simple expression evaluation
      // Note: Using eval is not recommended for production code
      // This is just for testing purposes
      const result = eval(this.display);
      this.display = result.toString();
    } catch (error) {
      this.display = "Error";
      throw error;
    }
  }

  memoryStore(): void {
    this.setValue("memory", parseFloat(this.display) || 0);
    this.clear();
  }

  memoryRecall(): void {
    const memoryValue = this.getValue("memory") || 0;
    this.display = memoryValue.toString();
  }

  memoryClear(): void {
    this.setValue("memory", 0);
  }

  memoryAdd(): void {
    const currentValue = parseFloat(this.display) || 0;
    const memoryValue = this.getValue("memory") || 0;
    this.setValue("memory", memoryValue + currentValue);
    this.clear();
  }

  handleSpecialButton(button: string): boolean {
    switch (button) {
      case "C":
        this.clear();
        return true;
      case "MS":
        this.memoryStore();
        return true;
      case "MR":
        this.memoryRecall();
        return true;
      case "MC":
        this.memoryClear();
        return true;
      case "M+":
        this.memoryAdd();
        return true;
      default:
        return false;
    }
  }

  press(button: string): Calculator {
    // Handle special buttons first
    if (this.handleSpecialButton(button)) {
      return this;
    }

    // For regular buttons, append to display
    this.display = this.display + button;
    return this;
  }

  getDisplay(): string {
    return this.display;
  }

  clear(): void {
    this.display = "";
  }

  // Keep these methods for backward compatibility if needed
  add(a: number, b: number): number {
    return a + b;
  }

  subtract(a: number, b: number): number {
    return a - b;
  }

  multiply(a: number, b: number): number {
    return a * b;
  }

  divide(a: number, b: number): number {
    if (b === 0) {
      throw new Error("Cannot divide by zero");
    }
    return a / b;
  }

  setValue(identifier: string, value: any): void {
    this.values[identifier] = value;
  }

  getValue(identifier: string): any {
    return this.values[identifier] ?? null;
  }
}

File: example/Calculator.py
class Calculator:
    """A simple calculator class for demonstration purposes."""
    
    def __init__(self):
        self.display = ""
        self.values = {}
        self.id = id(self)  # Add a unique ID to track instances

    def press(self, button: str):
        """Press a button on the calculator."""
        print(f"[CALCULATOR {self.id}] Pressing: {button}, current display: '{self.display}'")
        
        # Handle special buttons first
        if button == "C":
            return self.clear()
        if button == "MS":
            return self.memory_store()
        if button == "MR":
            return self.memory_recall()
        if button == "MC":
            return self.memory_clear()
        if button == "M+":
            return self.memory_add()
        
        # For regular buttons, append to display
        self.display = self.display + button
        print(f"[CALCULATOR {self.id}] New display is: '{self.display}'")
        return self

    def enter(self):
        """Evaluate the expression on the display."""
        try:
            # Simple expression evaluation using eval
            # Note: Using eval is not recommended for production code
            # This is just for testing purposes
            # pylint: disable=eval-used
            result = eval(self.display)
            self.display = str(result)
        except Exception as error:
            # We'll ignore the specific error for simplicity
            # pylint: disable=unused-variable
            self.display = "Error"
        return self

    def memory_store(self):
        """Store the current display value in memory."""
        try:
            value = float(self.display) if self.display else 0
            self.set_value("memory", value)
            self.clear()
        except ValueError:
            pass
        return self

    def memory_recall(self):
        """Recall the value from memory to the display."""
        memory_value = self.get_value("memory") or 0
        self.display = str(memory_value)
        return self

    def memory_clear(self):
        """Clear the memory value."""
        self.set_value("memory", 0)
        return self

    def memory_add(self):
        """Add the current display value to memory."""
        try:
            current_value = float(self.display) if self.display else 0
            memory_value = self.get_value("memory") or 0
            self.set_value("memory", memory_value + current_value)
            self.clear()
        except ValueError:
            pass
        return self

    def get_display(self):
        """Get the current display value."""
        print(f"[CALCULATOR {self.id}] getDisplay: '{self.display}'")
        return self.display

    def clear(self):
        """Clear the display."""
        self.display = ""
        return self

    # Keep these methods for compatibility
    def add(self, a, b):
        """Add two numbers."""
        return a + b

    def subtract(self, a, b):
        """Subtract b from a."""
        return a - b

    def multiply(self, a, b):
        """Multiply two numbers."""
        return a * b

    def divide(self, a, b):
        """Divide a by b."""
        if b == 0:
            raise ValueError("Cannot divide by zero")
        return a / b

    def set_value(self, identifier, value):
        """Set a value in the calculator's storage."""
        self.values[identifier] = value

    def get_value(self, identifier):
        """Get a value from the calculator's storage."""
        return self.values.get(identifier, None)
